bot/exts/halloween/halloweenify.py

```diff
@@ -37,11 +38,25 @@ async def halloweenify(self, ctx: commands.Context) -> None:
                 f"**{ctx.author.display_name}** wasn\'t spooky enough for you? That\'s understandable, "
                 f"{ctx.author.display_name} isn\'t scary at all! "
                 "Let me think of something better. Hmm... I got it!\n\n "
-                f"Your new nickname will be: \n :ghost: **{nickname}** :jack_o_lantern:"
             )
             embed.set_image(url=image)
 
-            await ctx.author.edit(nick=nickname)
+            try:
+                if isinstance(ctx.author, discord.Member):
+                    await ctx.author.edit(nick=nickname)
+                    embed.description += f"Your new nickname will be: \n:ghost: **{nickname}** :jack_o_lantern:"
+
+                else:   # The command has been invoked in DM
+                    embed.description += (
+                        f"Your new nickname should be: \n :ghost: **{nickname}** :jack_o_lantern: \n\n"
+                        f"Feel free to change it yourself, or invoke the command again inside the server."
+                    )
+
+            except Forbidden:   # The bot doesn't have enough permission
+                embed.description += (
+                    f"Your new nickname should be: \n :ghost: **{nickname}** :jack_o_lantern: \n\n"
+                    f"Although it looks like I can't change it myself, but feel free to change it yourself."
```

Comment:
`Although` or `but` should be removed to make this read well. Overall this could be written better, e.g.:
```
It looks like I cannot change your name, but feel free to change it yourself.
```